### PR TITLE
Forego full suite of CI builds for documentation-only changes

### PR DESCRIPTION
**Pull request by mattbjordan**
Both GitHub Actions and Azure Pipelines can be filtered to only run when certain files are changed.

- Skipped Azure Pipelines status checks resolve as neutral which is considered passing for required status checks.
- Skipped GitHub Actions status checks do not resolve at all, thus required status checks are left in an "expected" state. In this case, the status check is overridden by the new Skip Check workflow so that required GitHub Actions can be skipped but still recognized as passing.

### CODE REVIEW

**Comment by mattbjordan** [comment]
For an example showing that skipped Azure Pipelines still allow for the PR to be merged, see: https://github.com/mattbjordan/itwinjs-core-mergify/pull/41

For an example of how the status checks will look in itwinjs-core for a PR with only documentation changes, see: https://github.com/iTwin/itwinjs-core/pull/4212.

File: .github/workflows/extract-api.yaml
Comment (aruniverse) [inline .github/workflows/extract-api.yaml:14]: Looks good so far! Really excited to have this integrated. 

Can we also include the codeowners file?

File: .github/workflows/status-check-override.yaml
Comment (mattbjordan) [inline .github/workflows/status-check-override.yaml:49]: This - plus how the `non_documentation_files` variable is set below -  is not a great way of adding new files to be excluded...

I would like to investigate using a bash or python script (instead of or in addition to this paths-filter action) to check for only documentation files having been changed. I think that this can be in a separate PR though, so that the filtering can still be added now.

I imagine it would look something like:
```python
edited_files = [file.txt, docs/file.md]
edited_files_filtered = edited_files.filterOut("docs/**", "**.md", ...)

if edited_files_filtered is not empty:
	# there were non-documentation files edited
	dont_skip()
else:
	skip()
```

File: .github/workflows/extract-api.yaml
Comment (mattbjordan) [reply .github/workflows/extract-api.yaml:15]: I've added it to the excluded files. I tested it in my fork: https://github.com/mattbjordan/itwinjs-core-mergify/commit/bc2901aa2d76e8ea00dacab1bdb64ee72b6fd6d0. This commit shows that the status check skip is triggered and that the Azure Pipeline run is skipped after I made the same changes in the fork.

File: .github/workflows/status-check-override.yaml
Comment (skirby1996) [reply .github/workflows/status-check-override.yaml:49]: Hmm, maybe you could use [this action](https://github.com/marketplace/actions/changed-files) to get the full list of changed files, then you could probably use that to set your "documentation_files_only" variable in a single bash script.

Comment (mattbjordan) [reply .github/workflows/status-check-override.yaml:49]: That action looks great actually. I've been messing with it a bit and I am very likely going to switch to that. Nice find!

Comment (mattbjordan) [reply .github/workflows/status-check-override.yaml]: I've replaced the dorny/paths-filter action with this new one. The tj-actions/changed-files action has a built in boolean for checking if _only_ certain files have changed, which is exactly what I wanted.

File: .github/workflows/extract-api.yaml
Comment (mattbjordan) [reply .github/workflows/extract-api.yaml:15]: I replaced one of the actions I was using and now its easier to add/remove files to the exclusion list. The codeowners files is still added.

File: common/config/azure-pipelines/jobs/docs-ci.yaml
Comment (aruniverse) [inline common/config/azure-pipelines/jobs/docs-ci.yaml:22]: does this make sense? skipping the docs-ci pipeline when there are doc changes??

**Comment by aruniverse** [comment]
Why did the itwinjs ci pipeline get triggered from this pr? https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=1665400&view=results
there are no .ts?x changes in here, just pipeline changes

File: common/config/azure-pipelines/jobs/docs-ci.yaml
Comment (mattbjordan) [reply common/config/azure-pipelines/jobs/docs-ci.yaml:22]: Nope, didn't make sense. I've pushed a change so that the docs CI build will run when there are `.md` or `docs/` changes. I left in the exclusion of the codeowners file though. So if there's only a change to codeowners in a PR, it won't run. (I mostly left this in as a precedent for adding any other exclusions to this config in the future.)

File: .github/workflows/status-check-override.yaml
Comment (skirby1996) [inline .github/workflows/status-check-override.yaml:7]: Maybe we should make a note here that it is not necessary to fake the ADOps jobs that are skipped due to path filtering, and this is only needed for the checks ran in GH actions?

Comment (mattbjordan) [reply .github/workflows/status-check-override.yaml:7]: Good suggestion. How does the comment I just added sound?

**Review by calebmshafer — APPROVED** [review]
This looks good to me. After this merges please try it out with a handful of PRs to ensure it's working correctly in each of the cases.

**Review by skirby1996 — APPROVED** [review]

**Pull request merged**
